### DISCUSSION TITLE
ENOSPC: no space left on device, write on ~4.25gb

### DISCUSSION THREAD

**Discussion by 3kh0** [vercel/community#1293] [Help] [answered]
### Summary

Ok so I have a pretty large repo, it is a bit less than 4.25 gb and during deploy on vercel, we hit ENOSPC: no space left on device, write on after about 7 minutes of the deploy running, tried rerunning, yet nothing. Any help?

### Example

Sorry, the project is private and tons of people are trying to get the link 0.0

**Comment by proudparrot2** [comment]
3kh0 v4 moment

**Comment by Calandri** [comment]
Same here -> without any reason it stop deploying. No change was made on the repo. Error: ENOSPC: no space left on device, copyfile

**Comment by amyegan** [comment, marked as answer]
This error means your build was cancelled because it exceeded the disk space [limit](https://vercel.com/docs/concepts/limits/overview#:~:text=Custom-,Disk%20Size%20(GB),-13). 

> Sometimes, your Deployment Build can hit platform limits so that the build will be cancelled and throw a corresponding error that will be shown in the Build logs.
— [Cancelled Builds due to limits](https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#cancelled-builds-due-to-limits)

The size limit includes imported packages, icons, media files, etc. Be sure to review what you are importing in your code! Sometimes an installed package automatically updates to new version that is larger in size, and that can cause your build to become too large. You can avoid this by pinning specific version numbers or by using smaller or fewer packages.

**Reply by 3kh0** [reply]
How do i get a bigger limit?

**Reply by amyegan** [reply]
[Build container resource](https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#build-container-resources) limits cannot be increased and the same [disk size](https://vercel.com/docs/concepts/limits/overview#:~:text=Custom-,Disk%20Size%20(GB),-13) limit applies to all plans. 

The [build troubleshooting](https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#cancelled-builds-due-to-limits) doc suggests some things you can do to stay within the limit:
> - Review what package the error is related to and explore the package's documentation to see if the memory allocation can be customized with an install or Build command
> - If no package is specified, look into reducing the amount of JavaScript that your Project might be using or find a more efficient JavaScript bundler
> - Review what you are importing in your code such as third-party services, icons and media files
— [Cancelled Builds due to limits](https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#cancelled-builds-due-to-limits)

If you're using a CMS with Next.js or another compatible framework, you can use [Incremental Static Regeneration](https://vercel.com/docs/concepts/incremental-static-regeneration/overview#) to generate pages on-demand. This strategy can make builds smaller and faster for sites with a lot of content by deferring page generation until requested.

**Reply by 3kh0** [reply]
@amyegan We just have a static site, and our codebase only has things that are needed, any tips on that?

**Reply by amyegan** [reply]
I can only make general recommendations since I don't know your code. The links above have a lot of suggestions that help in most similar cases. 

If you're able to share your project, then we might be able to make specific suggestions on how to reduce the size of your build. But I understand your need to keep it private.

**Reply by 3kh0** [reply]
You can get a idea of it here: https://github.com/3kh0/3kh0.github.io/

It is mostly the same size, the new version removes alot of useless files that are there for some strange reason. We also made sure that we have (most) the images optimized for file size. I want to stay away from frameworks since they are a pain sometimes and I just want to keep things nice and simple.

**Reply by MSturges** [reply]
you ever find a solution?  I'm thinking of creating a ec2 instance using a ubuntu server with a lot of ram.

**Reply by amyegan** [reply]
There’s a hard 13GB limit to the available disk size used during builds, including Vercel's Build Cache. The `ENOSPC: no space left on device` error appears when the limit is exceeded. There are a couple of possible causes.

One possibility is a build that is simply too large. You can test for this by generating a build locally and then running `du -d 1 -h` at the root directory. 

A build cache issue is another possibility, such as a big change in `node_modules` dependencies. If you [redeploy](https://vercel.com/docs/deployments/managing-deployments#redeploy-a-project) a failed build without the build cache and it succeeds, then the build cache may be the problem. To avoid the build cache on new builds, you can [add an environment variable](https://vercel.com/docs/projects/environment-variables#declare-an-environment-variable) named `VERCEL_FORCE_NO_BUILD_CACHE` with value `1` to your project settings.

**Comment by HeleneGrini** [comment]
I am facing the same issue. My deploy stopped working with the same error. When i build locally the output size 1.7GB, which should be fine. I am able to deploy my site from my own computer using the cli. 

I have also tried using more ISR, although we dont have a whole lot of content, but the same error still occurs.  
And I have analysed my build pretty closely, without finding much 

@3kh0 were you able to get it up and running? 

@amyegan Do you have any ida what can cause this when its not actually the size of my app?

**Reply by 3kh0** [reply]
No, it never worked, sorry

**Comment by kodai3** [comment]
Hi, I am facing the similar issue. In my case, the error happens on the dependency install phase. 
If I deploy with `--force` or deploy without build cache from console (redeploy the failed build), it success.
with `du -d 1 -h` I get 3Gb which is way below the 13Gb limit.
```
 12K	./.scaffdog
120K	./config
2.0G	./node_modules
 32K	./.storybook
905M	./.next
 24K	./env
1.9M	./public
 36K	./.jest
 44K	./scripts
 12M	./src
2.9G	.
```

@amyegan 
Is it really not the problem of vercel's build or cache mechanism?
Is there a way to access the complete log from npm? I can't reproduce on my local env, so it is hard to narrow down the cause.


It happens frequently and blocks us from deploying the production (which we do everyday).

<img width="850" alt="image" src="https://github.com/vercel/community/assets/33568829/cb03659b-7d6f-45e1-8815-41fefb359487">

**Comment by seanhuggins1** [comment]
@kodai3 We are facing a similar issue during the install phase, some of our builds are starting to fail intermittently & redeploying fixes it.

This is the output from Vercel deployment details:
![image](https://github.com/vercel/community/assets/22649672/7e4bc703-40fc-499e-85d7-ad6052ba8ddd)

Similarily using `du -d 1 -h` we are below the 13Gb limit
![image](https://github.com/vercel/community/assets/22649672/66968b2b-2e30-43be-b10a-6539a011a301)

**Reply by kodai3** [reply]
Yes, It seems the same.
I asked to the vercel support team, but no response so far :(

**Reply by Tug** [reply]
Facing the same issue here this morning, I don't think my build has grown that much over night....
```
$  du -d 1 -h
4.5M	./plugins
3.1M	./node_modules
284K	./prisma
1.3G	./.next
6.9M	./public
116K	./scripts
892K	./src
1.3G	.
```

I'm in a monorepo though, this one is fairer:
```
du -d 1 -h ../..                                                                                                                                                        (base) 10:27:13
 16K	../../.husky
 28K	../../patches
1.9G	../../node_modules
 24M	../../packages
 52K	../../.github
 51M	../../.git
4.3G	../../apps
6.2G	../..
```

**Reply by nil1511** [reply]
Seems like swc packages are no longer ignored in tracing. 
Adding this to next.config.js should fix it for now
```js
experimental: {
          outputFileTracingIgnores: [
            // builds on vercel used to ignore these dependencies before but not anymore, so builds are failing
            // TODO remove these once builds are fixed in vercel
            '**/node_modules/@swc/core-linux-x64-gnu',
            '**/node_modules/@swc/core-linux-x64-musl',
            '**/node_modules/@esbuild/linux-x64',
            '**/node_modules/terser/dist',
            '**/node_modules/uglify-js/lib',
          ],
        },
```

**Reply by kodai3** [reply]
Related to [this](https://github.com/vercel/next.js/issues/42641) issue?

**Reply by nil1511** [reply]
yes, these dependencies started to appear from tomorrow, They are dev dependencies that should not be in serverless function.

**Reply by Tug** [reply]
> Adding this to next.config.js should fix it for now

Indeed it fixed it thanks 👍 

In our case we also needed to remove `outputFileTracing: false`, we had it in our config for [some reason](https://github.com/vercel/next.js/issues/30601#issuecomment-961323914).
